src/types/web3Provider.ts

```diff
@@ -0,0 +1,157 @@
+// This is a partial version of @web3-onboard/common's EIP-1193 type
+// https://github.com/blocknative/web3-onboard/blob/v2-web3-onboard-develop/packages/common/src/types.ts#L345
+// Importing the types via the module would be preferred, but this creates issues when generating a
+// rolled up .d.ts.
+// Specifically, the default rollup-plugin-ts uses the default tsc --declaration output,
+// which does not roll up module type definitions. rollup-plugin-dts only rolls up the immediately
+// imported modules, but does not roll-up nested/transitive modules, which these EIP-1193 types
+// have via the `eip-712` package.
```

Comment:
could we work around this by just directly importing the types from `@web3-onboard/common` somewhere so they make it into the final bundle types? or is the whole optional peer dependencies situation + this issue not worth doing it that way? Also i think this could be a `.d.ts` file

Reply:
IIRC that doesn't easily work because  just the types for `web3-onboard/common` refer to other dependencies it has, which yeah creates problems with the dependency installation path